confgenerator/confgenerator_test.go

```diff
@@ -172,18 +172,18 @@ func generateConfigs(platform platformConfig, testDir string) (got map[string]st
 			got["error"] = err.Error()
 		}
 	}()
-	// Merge Config
-	builtInConfBytes, confBytes, err := confgenerator.MergeConfFiles(
+
+	uc, err := confgenerator.MergeConfFiles(
 		filepath.Join("testdata", testDir, inputFileName),
 		platform.OS,
 		apps.BuiltInConfStructs,
 	)
 	if err != nil {
 		return
 	}
-	got[builtinConfigFileName] = string(builtInConfBytes)
+	got[builtinConfigFileName] = apps.BuiltInConfStructs[platform.OS].String()
 
-	uc, err := confgenerator.ParseUnifiedConfigAndValidate(confBytes, platform.OS)
+	err = uc.Validate(platform.OS)
```

Comment:
Every time `Validate` is called, or really any of the `UnifiedConfig` methods, we have to constantly pass in the platform. Maybe it should just be a private field?

Also maybe an aside, but while looking at this I noticed `uc.Logging.Validate` takes in a platform but doesn't use it. Also, I don't know why we have a `uc.ValidateMetrics` but also `uc.Logging.Validate` I don't understand why that API is different. Might be out of scope for this PR though.